core/src/main/java/org/apache/cxf/common/jaxb/JAXBUtils.java

```diff
@@ -1176,18 +1179,48 @@ private static Object createEscapeHandler(Class<?> cls, String simpleClassName)
         }
         return null;
     }
+    
+    @SuppressWarnings({ "deprecation", "removal" })
     public static JAXBContext createContext(final Set<Class<?>> classes,
                                             final Map<String, Object> map) throws JAXBException {
         JAXBContext ctx = null;
         try {
             ctx = AccessController.doPrivileged(new PrivilegedExceptionAction<JAXBContext>() {
                 public JAXBContext run() throws Exception {
-                    //This is a workaround for CXF-8675
-                    Class<?> factoryClass = ClassLoaderUtils.loadClass("org.glassfish.jaxb.runtime.v2.ContextFactory",
-                            JAXBContextCache.class);
-                    Method m = factoryClass.getMethod("createContext", Class[].class, Map.class);
-                    Object context = m.invoke(null, classes.toArray(new Class<?>[0]), map);
-                    return (JAXBContext) context;
+                    final String factoryClassName = System.getProperty(JAXBContext.JAXB_CONTEXT_FACTORY);
```

Comment:
@rmannibucau would appreciate if you validate the approach here, I think that would solve the underlying issue and properly instantiate the JAXB provider of choice, thanks in advance!

Reply:
Why not using first JAXBContext factory methods, now it is on jakarta they are there so reflection is not needed and all the new logic looks like the factoryfinder of the jaxb so can be worth avoiding to duplicate for cases it will be different (thinking to OSGi quite loud but I have other cases in mind like some tweaks in EE servers).

Reply:
Very valid point, it does look like factoryfinder, but something like should do it:

```
return JAXBContext.newInstance(classes.toArray(new Class<?>[0]), map);
```

Is that what you had in mind? thanks!